src/header/map.rs

```diff
@@ -638,6 +638,8 @@ impl<T> HeaderMap<T> {
 
         if cap > self.indices.len() {
             let cap = cap.next_power_of_two();
+            assert!(cap < MAX_SIZE, "header map reserve over max capacity");
+            assert!(cap != 0, "header map reserve overflowed");
```

Comment:
TIOLI: maybe [`usize::checked_next_power_of_two`](https://doc.rust-lang.org/std/primitive.usize.html#method.checked_next_power_of_two) + `expect` would express intent more clearly? Not a blocker.

Reply:
I find the asserts clearer, myself.